### PR TITLE
Coverity: code annotate false positives

### PR DESCRIPTION
**Pull request by c72578**
- Add code annotations according to:
  https://github.com/zeux/pugixml/issues/236

### CODE REVIEW

File: src/pugixml.cpp [hunk 1]
```diff
@@ -10863,6 +10865,7 @@ PUGI__NS_BEGIN
 			if (_next)
 				_next->optimize(alloc);
 
+			// coverity[var_deref_model]
```
Comment (zeux) [inline src/pugixml.cpp]: Is this required given the change https://github.com/zeux/pugixml/commit/d9fadc74646171d72b9a539ecff7183de1228f00?

Comment (c72578) [reply src/pugixml.cpp]: Commit https://github.com/zeux/pugixml/commit/d9fadc74646171d72b9a539ecff7183de1228f00 addressed the 4. issue in line 10926

Comment (c72578) [reply src/pugixml.cpp]: Maybe @cecilios can have a look and test with Coverity again

Comment (cecilios) [reply src/pugixml.cpp]: @c72578 Now quite bussy but I will check on this weekend.

**Comment by cecilios** [comment]
I have checked with Coverity: It complains in two lines, 10928 and 10087. It seems that commit d9fadc7 didn't solve the issue in line 10928.
I will add a coverity annotation in 10928 and move the current one near 10087 and re-check

**Comment by c72578** [comment]
Closed this PR in favor of https://github.com/zeux/pugixml/pull/241

**Pull request closed without merging**